scripts/loadable_app/hello.c

```diff
@@ -0,0 +1,12 @@
+int a = 9;
+int b = 9;
+
+int x()
+{
+return 7;
+}
+int main(int argc, char **argv)
+{
+	int p = x();
+	return a + p + x();
```

Comment:
Interesting, binaries are definitely different.
My compiler turned out to make proper binary `gcc -static -m32 -emain -nostartfiles -fpic hello.c -c -o hello.o`
```
00000000 <main>:
   0:   55                      push   %ebp
   1:   89 e5                   mov    %esp,%ebp
   3:   e8 fc ff ff ff          call   4 <main+0x4>
   8:   05 01 00 00 00          add    $0x1,%eax
   d:   8b 80 00 00 00 00       mov    0x0(%eax),%eax
  13:   8b 00                   mov    (%eax),%eax
  15:   5d                      pop    %ebp
  16:   c3                      ret    

00000000 <__x86.get_pc_thunk.ax>:
   0:   8b 04 24                mov    (%esp),%eax
   3:   c3                      ret    
```
That is changed to wrong one during linking, and I got what I've got. 

Could you find out gcc options to work with https://github.com/embox/crosstool/releases/download/2.28-6.3.0-7.12/i386-elf-toolchain.tar.bz2? Or https://github.com/embox/crosstool/releases/download/2.25.1-4.9.2-7.9/i386-elf-toolchain.tar.bz2. So we could bypass headache with host compiler. Sadly, it doesn't work out of box